omise-php/lib/omise/res/obj/OmiseObject.php

```diff
@@ -20,7 +20,14 @@ protected function __construct($publickey = null, $secretkey = null) {
       $this->_publickey = OMISE_PUBLIC_KEY;
     }
     if($secretkey !== null) {
-      $this->_secretkey = $secretkey;
+
+      // I don know why, but there is a pattern that is '$secretkey == $publickey'.
+      if($secretkey == $publickey){
```

Comment:
Hi! Thank you for your valuable contribution! Appreciate your commitment.
However this line here seems very strange to me. I would say need to test. Actually secret key should never be equal to public key. And I am not sure if this condition is to validate user mistake or what. I would say need to verify charge creation for this condition. May I know if you have a chance to do e2e testing for this?

Reply:
> May I know if you have a chance to do e2e testing for this?
Yes, of course.
Is it ok to attach the video about e2e testing?

Reply:
@shunsuke-h-opn I'm also curious about this. Can you add some more detail about why you found it necessary to add?

Reply:
@danfowler I will add some more details in the beginning next week.
As you know, it's very strange situation, but it happened when charge create.
But I need to prepare some debug log. please wait.
11/3 is public holiday in Japan.

Reply:
@danfowler @annatotest 
Sorry, I could fix this problem.
https://github.com/omise/omise-eccube/pull/14/commits/1adaf99536c73fe55faec21244f7dbc2e46e8e87